src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,296 @@
+package duke;
+
+import duke.exception.DescriptionFieldEmptyException;
+import duke.exception.MultipleTimeFieldsException;
+import duke.exception.TimeFieldEmptyException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Duke {
+
+    public static void main(String[] args) {
+
+        Scanner userInput = new Scanner(System.in);
+
+        printStartingMessage();
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        Task task;
+        String[] description;
+        String inputLine;
+        int COMMAND_TASK_SEPARATOR = 2;
+        String filePath = new File("").getAbsolutePath();
+
+        try {
+            tasks = loadfromFile(filePath + "/duke.txt");
+        } catch (FileNotFoundException e) {
+            System.out.println(" I've created a text file at " + filePath + " to save your tasks!");
+            printHorizontalLine();
+        }
+
+        while (!(inputLine = userInput.nextLine().trim()).equals("bye")) {
+
+            printHorizontalLine();
+
+            String[] command = inputLine.split(" ", COMMAND_TASK_SEPARATOR);
+
+            switch (command[0].toLowerCase()) {
+            case "list":
+                printListMessage(tasks);
+                break;
+            case "done":
+                try {
+                    int doneTaskNumber = Integer.parseInt(command[1]);
+                    task = tasks.get(doneTaskNumber - 1);
+                    task.markAsDone();
+                    printDoneMessage(task);
+                } catch (IndexOutOfBoundsException e) {
+                    printNonExistentTaskMessage();
+                } catch (NumberFormatException e) {
+                    printNotANumberMessage();
+                }
```

Comment:
The two cases have varying levels of abstraction, consider standardizing the level to SLAP. For example, the contents of the `done` case can be extracted to a separate method to mark the task as done. Perhaps the same could be done for the other case statements as well? This could reduce the size of the `main` method considerably.